api-documentation/payments.md

```diff
@@ -1,102 +1,445 @@
+`country_code` | `NL`
+`external_invoice_number` | `123456`
+`amount_cents` | `1000`
+`last_name` | `Doe`
+`locale` | ``
+`company_id` | `d4b8772c67154a6bced8a8b827e177cc00111fe0`
+`payment_reference` | `Club membership 2019/2`
+
+partner API key: `3ac2bf2359c1eb184fe0fea01f624bc1d8581981`
+
+1. Discard key-value pairs for which value is null or an empty string
+
+key | value
+--- | ---
+`first_name` | `John`
+`redirect_url` | `http://partner-test.nl`
+`country_code` | `NL`
+`external_invoice_number` | `123456`
+`amount_cents` | `1000`
+`last_name` | `Doe`
+`company_id` | `d4b8772c67154a6bced8a8b827e177cc00111fe0`
+`payment_reference` | `Club membership 2019/2`
+
+2. Sort the key-value pairs by key.
+
+key | value
+--- | ---
+`amount_cents` | `1000`
+`company_id` | `d4b8772c67154a6bced8a8b827e177cc00111fe0`
+`country_code` | `NL`
+`external_invoice_number` | `123456`
+`first_name` | `John`
+`last_name` | `Doe`
+`payment_reference` | `Club membership 2019/2`
+`redirect_url` | `http://partner-test.nl`
+
+3. Concatenate every key-value pair on the list obtained in the previous step to get the signing string
+
+```
+signing_string = params.keys.sort.flat_map { |k| [k, params[k]] }.join
+```
+
+```
+amount_cents1000company_idd4b8772c67154a6bced8a8b827e177cc00111fe0country_codeNLexternal_invoice_number123456first_nameJohnlast_nameDoepayment_referenceClub membership 2019/2redirect_urlhttp://partner-test.nl
+```
+
+4. Calculate SHA256 digest of the signing string
+
+```
+digested_content = Digest::SHA256.digest(signing_string)
+```
+
+```
+\tD!()\xAFL\xED\x069\xC0\xEC\x11/\xB3\x93\xFAM\xC0\xA9\xE7\x03\x03\e\xDC\xDFF\x9FZ\x15\xD6\xD1
+```
+
+5. Calculate HMAC SHA-256 signature of digested signing string using the API key.
+
+```
+hmac_binary = OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), api_key, digested_content)
+```
+
+6. Encode the result from binary to hexadecimal
+```
+signature = hmac_binary.unpack1('H*')
+```
+
+The signature calculated for the example key-value pairs and API key is:
+```
+754966cc8946c8125b365fcb5cf0e27edd98fe7516de7ea17f1b5254bcf7a00e
+```
+
+After the signature is generated, it has to be appended to the url query string as
+`&signature=754966cc8946c8125b365fcb5cf0e27edd98fe7516de7ea17f1b5254bcf7a00e`
+
+##Payment response##
+
+Once the users finish the payment process, they are redirected to a result page of your choice, included in the payment request as `redirect_url`.
+
+ClubCollect will append parameters to this URL. `payment_result` will inform the partner about the payment status. If the status is already determined (either `authorised` or `refused` or `cancelled`), this information can be used to display a payment successful or payment failed or payment cancelled page.
+
+In a case when the current status is `pending`, the outcome of the payment will be communicated with payment notifications. Payments with status `cancelled` may be authorized later by the payment provider.
```

Comment:
Here I would leave a reference saying that the cancelled payments might be authorised later via the payments api webhook.